python/openvino_tokenizers/tokenizer_pipeline.py

```diff
@@ -839,18 +839,30 @@ def get_ov_subgraph(self, input_nodes):
 
         op_inputs = []
         input_nodes_iter = iter(input_nodes)
-        for node in self.inputs:
-            if isinstance(node, Sequence):
-                op_inputs.extend(islice(input_nodes_iter, 3))
-            elif isinstance(node, AddToken):
-                # Put a scalar as a ragged tensor with scalar shape and a single element
+
+        segment_ids = []
+        segment_index = 0
+        for key, group_iter in groupby(self.inputs, key=type):
+            if key is Sequence:
+                for sequence in group_iter:
+                    op_inputs.extend(islice(input_nodes_iter, 3))
+                segment_ids.append(self.segment_ids[segment_index])
+                segment_index += 1
+            elif key is AddToken:
+                ids = [node._token_id for node in group_iter]
+                
+                tmp_seg_ids = np.array(self.segment_ids[segment_index:segment_index + len(ids)])
+                assert (tmp_seg_ids == tmp_seg_ids[0]).all()
+                segment_ids.append(self.segment_ids[segment_index])
+                segment_index += len(ids)
+
                 op_inputs.extend(make_constant_node(0, Type.i32).outputs())
-                op_inputs.extend(make_constant_node(1, Type.i32).outputs())
-                op_inputs.append(make_constant_node(np.array([node._token_id]), Type.i32).output(0))
+                op_inputs.extend(make_constant_node(len(ids), Type.i32).outputs())
+                op_inputs.append(make_constant_node(np.array(ids), Type.i32).output(0))
             else:
-                raise UserInputError(f"Unexpected node type in CombineSegments: {type(node)}")
+                raise UserInputError(f"Unexpected node type in CombineSegments: {key}")
```

Comment:
```suggestion
                raise UserInputError(f"Unexpected node type in CombineSegments: {key[0]}")
```

Reply:
`key` is a variable holding Type, it cannot be sliced.
![image](https://github.com/user-attachments/assets/880e1a71-6dc9-47e1-95c4-33cece2c87c3)

Reply:
Now, i see. We should change lambda as well

Reply:
no need to do `{key[0]}` if for iteration is done the following way
`for (key, token_id), group_iter in groupby(self.inputs, key=lambda input: (type(input), getattr(input, 'token_id', None))):`